File: VB/DataGridWinApp1/Form1.vb
Imports System
Imports System.ComponentModel
Imports System.Drawing
Imports System.Windows.Forms
Imports DevExpress.XtraGrid.Views.Grid

Namespace DataGridWinApp1

    Public Partial Class Form1
        Inherits Form

        Private data As BindingList(Of DataClass) = New BindingList(Of DataClass)()

        Private gifHelper As AnimatedGifImageHelper = New AnimatedGifImageHelper("..\..\star_icons04.gif")

        Private imgToDraw As Image = Nothing

        Public Sub New()
            InitializeComponent()
        End Sub

        Private Sub Form1_Load(ByVal sender As Object, ByVal e As EventArgs)
            For i As Integer = 0 To 20 - 1
                data.Add(New DataClass(i))
            Next

            dataClassBindingSource.DataSource = data
        End Sub

        Private Sub timer1_Tick(ByVal sender As Object, ByVal e As EventArgs)
            imgToDraw = gifHelper.GetNextFrame()
            gridView1.InvalidateRowIndicator(gridView1.FocusedRowHandle)
        End Sub

        Private Sub gridView1_CustomDrawRowIndicator(ByVal sender As Object, ByVal e As RowIndicatorCustomDrawEventArgs)
            Dim view As GridView = CType(sender, GridView)
            If e.RowHandle IsNot view.FocusedRowHandle OrElse Not view.IsDataRow(e.RowHandle) OrElse imgToDraw Is Nothing Then Return
            e.Info.ImageIndex = -1
            Dim imgLocation As Point = New Point()
            imgLocation.Y = e.Bounds.Y + e.Bounds.Size.Height / 2 - imgToDraw.Size.Height / 2
            imgLocation.X = e.Bounds.X + e.Bounds.Size.Width / 2 - imgToDraw.Size.Width / 2
            e.Painter.DrawObject(e.Info)
            e.Graphics.DrawImage(imgToDraw, imgLocation)
            e.Handled = True
        End Sub
    End Class
End Namespace

File: CS/DataGridWinApp1/Form1.cs
using System;
using System.ComponentModel;
using System.Drawing;
using System.Windows.Forms;
using DevExpress.XtraGrid.Views.Grid;

namespace DataGridWinApp1
{
	public partial class Form1 : Form
	{
		private BindingList<DataClass> data = new BindingList<DataClass>();
		private AnimatedGifImageHelper gifHelper = new AnimatedGifImageHelper("..\\..\\star_icons04.gif");
		private Image imgToDraw = null;

		public Form1()
		{
			InitializeComponent();
		}

		private void Form1_Load(object sender, EventArgs e)
		{
			for ( int i = 0; i < 20; i++ )
				data.Add(new DataClass(i));

			dataClassBindingSource.DataSource = data;
		}

		private void timer1_Tick(object sender, EventArgs e)
		{
			imgToDraw = gifHelper.GetNextFrame();
			gridView1.InvalidateRowIndicator(gridView1.FocusedRowHandle);
		}

		private void gridView1_CustomDrawRowIndicator(object sender, RowIndicatorCustomDrawEventArgs e)
		{
			GridView view = (GridView)sender;
			if ( e.RowHandle != view.FocusedRowHandle || !view.IsDataRow(e.RowHandle) || imgToDraw == null )
				return;

			e.Info.ImageIndex = -1;
			Point imgLocation = new Point();
			imgLocation.Y = e.Bounds.Y + e.Bounds.Size.Height / 2 - imgToDraw.Size.Height / 2;
			imgLocation.X = e.Bounds.X + e.Bounds.Size.Width / 2 - imgToDraw.Size.Width / 2;

			e.Painter.DrawObject(e.Info);
			e.Graphics.DrawImage(imgToDraw, imgLocation);
			e.Handled = true;
		}
	}
}